charm/kafka/config.yaml

```diff
@@ -16,6 +16,12 @@ options:
     description: |-
       Default replication factor for automatically created topics
       (default.replication.factor)
+  log4j_properties:
+    default: ""
+    type: string
+    description: |-
+      Contents of a custom log4j.properties file for the Kafka brokers. If
```

Comment:
What is the default? From the code below, it looks like if it's empty and the file exists, it will be removed. Otherwise, if it's empty and the file does not exist, nothing happens.

Reply:
The changes to the snap wrapper will cause the custom log4j.properties file to be used if it exists -- otherwise the kafka startup script will use a default configuration, `/opt/kafka/config/log4j.properties`.

So we want to remove that custom file if a custom log4j_properties was set in the past and a custom log4j.properties file was written, but it is no longer set (config change from non-empty string to empty string, `juju config --reset kafka log4j_properties`, etc)